### PR TITLE
.github: Don't persist credentials in repository

### PR DESCRIPTION
**Pull request by pchaigno**
When using actions/checkout, the default behavior [1] is to persist git credentials in the checked out code. This is ill-advised, so let's disable with `persist-credentials`.

1 - https://github.com/actions/checkout#usage
2 - https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
SGTM. Do we need to test these before we merge them to make sure we're not relying on them?

**Comment by pchaigno** [comment]
> Do we need to test these before we merge them to make sure we're not relying on them?

I'd prefer if we can, yes. But it seems GitHub workflows are on strike in this PR...

File: .github/workflows/bpf-checks.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
+        persist-credentials: false
```
Comment (nbusseneau) [inline .github/workflows/bpf-checks.yaml]: `persist-credentials` should be within the `with` block with the other arguments ;)

**Review by nbusseneau — CHANGES_REQUESTED** [review]

**Comment by nbusseneau** [comment]
> I'd prefer if we can, yes. But it seems GitHub workflows are on strike in this PR...

This is because of the syntax error outlined above: https://github.com/cilium/cilium/actions/runs/821429393

File: .github/workflows/bpf-checks.yaml (hunk 1 above)
Comment (pchaigno) [reply .github/workflows/bpf-checks.yaml]: :facepalm:

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ jobs:
     steps:
     - name: Checkout repo
       uses: actions/checkout@v2
+      persist-credentials: false
```
Comment (nbusseneau) [inline .github/workflows/codeql-analysis.yml]: Missed this one :D

**Review by nbusseneau — CHANGES_REQUESTED** [review]

**Review by nbusseneau — APPROVED** [review]
Otherwise LGTM.

**Comment by pchaigno** [comment]
Team reviews are in and all GitHub workflows are passing. Marking as ready to merge.

**Pull request merged**